fix(ivy): compile interpolated bindings without bind instruction

JoostK opened the pull request:
This fixes the case where the compiler would generate a bind(interpolation#())
instruction.

## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

In #23881 it was discovered that the compiler may generate an `interpolation#` instruction within the `bind` instruction. This sequence of instructions will fail a runtime assertion once #23881 is merged, given that by then a potential `NO_CHANGE` value from `interpolation#` is no longer accepted by `bind`.

## What is the new behavior?

The compiler correctly identifies the interpolated binding and does not generate the `bind` instruction in that case.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

For the scope of this bugfix I have not included testcases for bindings in the form of `<a title={{ title }}></a>`, given that this would currently generate `interpolation0('', ctx.title, '')` which is again not necessarily the output we want the compiler to generate, as we can optimize to `bind`.

JoostK committed:
fix(ivy): compile interpolated bindings without superfluous bind instruction

This fixes the case where the compiler would generate a bind(interpolation#())
instruction.

vicb commented on packages/compiler/test/render3/r3_view_compiler_binding_spec.ts:
```diff
@@ -0,0 +1,126 @@
+      };
+
+      const template = `
+      // ...
```
not needed at the beginning (the expectation is not anchored)

vicb reviewed (approved):
LGTM, some minor cleanup required.

vicb commented:
http://test/OCL:196742201:BASE:196742230:1526424224848:919479a1

JoostK committed:
fixup! fix(ivy): compile interpolated bindings without superfluous bind instruction

This fixes the case where the compiler would generate a bind(interpolation#())
instruction.

vicb commented:
Thanks for the updates @JoostK

matsko commented:
@JoostK please rebase...

JoostK commented:
@matsko Done!
